coredhcp.go

```diff
@@ -187,11 +190,21 @@ func (s *Server) MainHandler4(conn net.PacketConn, peer net.Addr, req *dhcpv4.DH
 	}
 
 	if resp != nil {
+		switch mt := req.MessageType(); mt {
```

Comment:
I would move this a little bit up, to line 180, right after `NewReplyFromRequest` and before calling all the handlers.